File: AN_2/SEMESTRUL_IV/MPP/LABORATOR/CSHARP_TEMA/tema_sapt3/Networking/servers/RpcConcurrentServer.cs
using System;
using System.Net.Sockets;
using System.Threading;
using networking.rpcprotocol;
using networking.servers;
using services;
using tema_bun.AppProtobuffs;

public class RpcConcurrentServer : AbsConcurrentServer
{
    private IServices server;
    
    public RpcConcurrentServer(int port,IServices s) : base(port)
    {
        server=s;
        Console.WriteLine("RpcConcurrentServer");
    }

    protected override Thread createWorker(TcpClient client)
    {
        ClientRpcWorker worker = new ClientRpcWorker(server, client);
        //ProtobuffWorker worker = new ProtobuffWorker(server, client);
        Thread t = new Thread(() => worker.run());
        return t;
    }
    
    public void stop(){Console.WriteLine("Stopping services...");}
}

File: AN_2/SEMESTRUL_IV/MPP/LABORATOR/JAVA_TEMA/tema_bun/Networking/src/main/java/com/example/tema_bun/servers/RpcConcurrentServer.java
package com.example.tema_bun.servers;


import com.example.tema_bun.IServices;
import com.example.tema_bun.objectProtocol.ClientRpcWorker;

import java.net.Socket;

public class RpcConcurrentServer extends AbsConcurrentServer{

    private IServices server;

    public RpcConcurrentServer(int port, IServices s) {
        super(port);
        this.server = s;
        System.out.println("RpcConcurrentServer");
    }

    @Override
    protected Thread createWorker(Socket client) {
        ClientRpcWorker worker = new ClientRpcWorker(server, client);
        Thread t = new Thread(worker);
        return t;
    }

    @Override
    public void stop(){
        System.out.println("Stopping services ...");
    }
}
